File: sample/src/main/java/me/kaelaela/sample/MainActivity.java
package me.kaelaela.sample;

import android.net.Uri;
import android.os.Bundle;
import android.support.customtabs.CustomTabsIntent;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.SeekBar;
import me.kaelaela.opengraphview.OpenGraphView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final OpenGraphView openGraphView = (OpenGraphView) findViewById(R.id.og_view);
        if (openGraphView != null) {
            openGraphView.loadFrom("http://ogp.me/");
            openGraphView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    final CustomTabsIntent intent = new CustomTabsIntent.Builder().setShowTitle(true).build();
                    intent.launchUrl(MainActivity.this, Uri.parse(((OpenGraphView) v).getUrl()));
                }
            });
        }
        findViewById(R.id.button1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openGraphView.clear();
                openGraphView.loadFrom("http://ogp.me/");
            }
        });
        findViewById(R.id.button2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openGraphView.clear();
                openGraphView.loadFrom("https://about.me/kaelaela");
            }
        });
        findViewById(R.id.button3).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openGraphView.clear();
                openGraphView.loadFrom("https://github.com/kaelaela");
            }
        });
        findViewById(R.id.button4).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openGraphView.clear();
                openGraphView.loadFrom("http://blog.kaelae.la/");
            }
        });

        Button loadButton = (Button) findViewById(R.id.load_button);
        final EditText editText = (EditText) findViewById(R.id.url_form);
        editText.setSelection(7);
        if (loadButton != null) {
            loadButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    openGraphView.clear();
                    openGraphView.loadFrom(String.valueOf(editText.getText()));
                    editText.setText("http://");
                    editText.setSelection(7);
                }
            });
        }

        findViewById(R.id.left_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openGraphView.setImagePosition(OpenGraphView.IMAGE_POSITION.LEFT);
            }
        });
        findViewById(R.id.right_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openGraphView.setImagePosition(OpenGraphView.IMAGE_POSITION.RIGHT);
            }
        });

        final float density = getResources().getDisplayMetrics().density;
        ((SeekBar) findViewById(R.id.radius_seek_bar)).setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                openGraphView.setCornerRadius(progress * density);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {

            }
        });

        findViewById(R.id.open_list_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                LinkListActivity.launch(v.getContext());
            }
        });
    }
}

File: sample/src/main/java/me/kaelaela/sample/MainActivity.kt
package me.kaelaela.sample

import android.net.Uri
import android.os.Bundle
import android.support.customtabs.CustomTabsIntent
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.SeekBar
import me.kaelaela.opengraphview.OnLoadListener
import me.kaelaela.opengraphview.OpenGraphView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val openGraphView = findViewById(R.id.og_view) as OpenGraphView
        //You can change the original parser by following code.
        //        openGraphView.setCustomParser(new XMLPullSampleParser());
        //        openGraphView.setCustomParser(new JSoupSampleParser());
        //        openGraphView.setCustomParser(new JSpoonSampleParser());
        openGraphView.setOnLoadListener(object : OnLoadListener() {
            override fun onLoadError(e: Throwable) {
                Log.d("TAG", e.message)
                openGraphView.visibility = View.GONE
            }
        })
        openGraphView.loadFrom("http://ogp.me/")
        openGraphView.setOnClickListener { v ->
            val intent = CustomTabsIntent.Builder().setShowTitle(true).build()
            intent.launchUrl(this@MainActivity, Uri.parse((v as OpenGraphView).url))
        }

        findViewById(R.id.open_list_button).setOnClickListener { v -> LinkListActivity.launch(v.context) }
        findViewById(R.id.button1).setOnClickListener {
            openGraphView.clear()
            openGraphView.loadFrom("http://ogp.me/")
        }
        findViewById(R.id.button2).setOnClickListener {
            openGraphView.clear()
            openGraphView.loadFrom("https://about.me/kaelaela")
        }
        findViewById(R.id.button3).setOnClickListener {
            openGraphView.clear()
            openGraphView.loadFrom("https://github.com/kaelaela")
        }
        findViewById(R.id.button4).setOnClickListener {
            openGraphView.clear()
            openGraphView.loadFrom("http://blog.kaelae.la/")
        }

        val loadButton = findViewById(R.id.load_button) as Button
        val editText = findViewById(R.id.url_form) as EditText
        editText.setSelection(7)
        loadButton.setOnClickListener {
            openGraphView.clear()
            openGraphView.loadFrom(editText.text.toString())
            editText.setText("http://")
            editText.setSelection(7)
        }

        findViewById(R.id.left_button).setOnClickListener { openGraphView.setImagePosition(OpenGraphView.IMAGE_POSITION.LEFT) }
        findViewById(R.id.right_button).setOnClickListener { openGraphView.setImagePosition(OpenGraphView.IMAGE_POSITION.RIGHT) }

        val density = resources.displayMetrics.density
        (findViewById(R.id.radius_seek_bar) as SeekBar).setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                openGraphView.setCornerRadius(progress * density)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {

            }

            override fun onStopTrackingTouch(seekBar: SeekBar) {

            }
        })
    }
}
